shell.nix

```diff
@@ -1,17 +1,14 @@
+# Note: Passing arguments through here to customize the environment does not
+# work on Nix 2.3.  It works with Nix 2.5.  I'm not sure about 2.4.
```

Comment:
I guess this and `tests.nix` aren't consumed by `PrivateStorageio` so requiring nix 2.4+ isn't an issue for them.